### PR TITLE
Add ability to index on more than one cell if desired

### PR DESCRIPTION
**Pull request by dwightguth**
**please review**

### CODE REVIEW

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
I'm not entirely certain why, but apparently the problem I was having, that unindexed rules were a performance bottleneck, goes away with this diff even if I don't actually index on more than the K cell. I'm inclined to try to commit this though, since it seems to have a roughly 40% speed increase on some programs in the C semantics. If we commit it and run into bugs or failed tests or loss in performance, I'll investigate further.

**Comment by dwightguth** [comment]
@yilongli please review

**Comment by yilongli** [comment]
lgtm

**Comment by dwightguth** [comment]
The problem is that the indexingPair method is still used by stream cells, and I'm not entirely sure how to get the cell associated with those rules, which would be needed in order to store it in the map.

**Comment by yilongli** [comment]
OK, then I am fine with it.

**Comment by dwightguth** [comment]
Jenkins: test this please

File: java-backend/src/main/java/org/kframework/backend/java/indexing/IndexingTable.java [hunk 1]
```diff
@@ -29,16 +40,16 @@
 /**
  * The indexing scheme currently used in the Java Backend
  */
-public class IndexingTable implements Serializable, RuleIndex {
+public class IndexingTable implements Serializable {
 
-    private Map<Index, ImmutableSet<Rule>> ruleTable;
+    private transient Table<CellLabel, Index, ImmutableSet<Rule>> ruleTable;
```
Comment (andreistefanescu) [inline java-backend/src/main/java/org/kframework/backend/java/indexing/IndexingTable.java:45]: why is this transient?

Comment (dwightguth) [inline java-backend/src/main/java/org/kframework/backend/java/indexing/IndexingTable.java:45]: not sure... seems like a bug. sorry

**Pull request closed without merging**